R/derive_params_growth_age.R

```diff
@@ -162,6 +173,7 @@
 #'
 #' derive_params_growth_age(
 #'   advs,
+#'   by_vars = exprs(STUDYID, USUBJID, VISIT),
```

Comment:
@Fanny-Gautier @rossfarrugia alternatively this can be rewritten to something like `c(admiral::get_admiral_options("subject_keys"), exprs(VISIT)`

Reply:
The function examples don't really matter and fine to include USUBJID there. It's only within function and template code that we really need to avoid hardcoding `STUDYID` and `USUBJID` - otherwise if user sets different options here like `STUDYID2` and `USUBJID2` then these 2 don't get consistently used in all their code.

Reply:
I saw you updated the other `by_vars` example calls to `get_admiral_options("subject_keys")` - I would just put those back to `exprs(STUDYID, USUBJID)` if you keep this one as it is, as the main thing is the examples should be consistent. I imagine many users don't maintain use of `get_admiral_options("subject_keys")` in the admiral code that they add too, as most probably never have a need to use anything other than `STUDYID` and `USUBJID`

Reply:
@zdz2101 Did you try to run any code while `by_vars = NULL` ?
I get the following error message:
![image](https://github.com/pharmaverse/admiralpeds/assets/157114584/b96a29d9-b771-416c-8dd3-009b4cd9d276)

Reply:
i'm not sure if we usually accept `by_vars=NULL` in this kind of function and it still runs? maybe double check https://github.com/pharmaverse/admiral/blob/main/R/derive_extreme_event.R.

i do think we should update our by_vars argument documentation though to be more consistent with admiral core for all our derive functions including the new interpolation one particularly calling out what values we expect here like below:

![image](https://github.com/pharmaverse/admiralpeds/assets/82581364/f245f6ba-aa99-42b5-9a80-325e035af6cf)

Reply:
On further discussion over chat with Fanny, i think the interpolation function is fine to add extra IF steps so that if by_vars=NULL then it still works with no groupings applied, as you don't always need groupings there. 

@zdz2101 for your functions I think it's key to add to the documentation that a list of variables created by exprs() **is expected** here for by_vars